### ISSUE TITLE
node-lazy is not compatible with NodeJS 0.10.5

### ISSUE THREAD

**Issue by javiergarmon** [pkrumins/node-lazy#35]
Hi!

In 0.10.5 version of NodeJS  there is a change with the streams:
- stream: Fix unshift() race conditions (isaacs)

So now node-lazy is not compatible with this version, this is the error:

```
/usr/lib/node_modules/forever/node_modules/nssocket/node_modules/lazy/lazy.js:211
            for (var i = 0; i < chunk.length; i++) {
                                     ^
TypeError: Cannot read property 'length' of null
    at Function.<anonymous> (/usr/lib/node_modules/forever/node_modules/nssocket/node_modules/lazy/lazy.js:211:38)
    at Lazy.<anonymous> (/usr/lib/node_modules/forever/node_modules/nssocket/node_modules/lazy/lazy.js:187:21)
    at Lazy.EventEmitter.emit (events.js:95:17)
    at Socket.<anonymous> (/usr/lib/node_modules/forever/node_modules/nssocket/node_modules/lazy/lazy.js:50:22)
    at Socket.EventEmitter.emit (events.js:92:17)
    at emitReadable_ (_stream_readable.js:392:10)
    at emitReadable (_stream_readable.js:388:5)
    at Socket.Readable.read (_stream_readable.js:250:5)
    at Socket.onSocketEnd (net.js:256:10)
    at Socket.EventEmitter.emit (events.js:92:17)
```

I've checked all the modules and dependecies and they are updated, so I think the error is cause by this NodeJS update.

Thanks!

**Comment by arch1t3ct** [comment]
Same here. Needs a fix.

**Comment by PaulFreund** [comment]
Same here

**Comment by pianomansam** [comment]
Issue is due to var chunk sometimes being null. As null, it doesn't have a property of ".length", which is assumed in line 211 and 231. There might be a deeper cause to this, but we can simply check for null in these two places before attempting to check length.

This is my first time contributing to an issue on github, so I might not be doing this correctly, but here's a diff file that does the checking for null. http://cl.ly/code/2k1I3A2r2v0k

**Comment by perezpaya** [comment]
@pianomansam Thank you for your diff, we (@javiergarmon and I) have fixed our issue with Lazy and Forever :P

**Comment by pkrumins** [comment]
Patch merged and published as 1.0.11.
